### PR TITLE
feat(docs): add API proposals for scheduler

### LINKED ISSUE #9: Define scheduler API
### What is the problem you're trying to solve?

We are tying to define an API for the scheduler module.
The API should give ways for the node agent and controller module to easily create or delete instances and nodes.

### Task list

- [x] Define `.proto` file
- [x] Write documentation

### Anything else ?

_No response_

### PR DESCRIPTION
**Pull request by iverly**
## Related issues :
Close #9 

## Overall explanation of your work :

We have added the documentation and our `.proto` file for scheduler's APIs.

## Technical decisions you've made and why :

Not necessary.

## Tasks lists :

- [x] Define `.proto` file
  - [x] Define structures
  - [x] Define enumerations
  - [x] Define services
- [x] Write documentation

### CODE REVIEW

File: docs/proposals/scheduler/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+message InstanceAction {
+	string id = 1;
+	Action action = 2;
+}
+```
+
+```protobuf
+// Represents a Node Register request
+message NodeRegisterRequest {
+	string certificate = 1;
```
Comment (iverly) [inline docs/proposals/scheduler/README.md:75]:
```suggestion
	string certificate = 2;
```

File: docs/proposals/scheduler/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+### Structures
+
+---
+
+```protobuf
+// Represents an Instance (eg. a container, VM ...)
+message Instance {
+    string id = 1;
+    string name = 2;
```
Comment (Alexis-Bernard) [inline docs/proposals/scheduler/README.md:21]:
```suggestion
    string name = 1;
```

File: docs/proposals/scheduler/scheduler.proto [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+message NodeRegisterRequest {
+    string certificate = 1;
+}
+
+message NodeRegisterResponse {
+    int code = 1;
+    string description = 2;
+}
```
Comment (Alexis-Bernard) [inline docs/proposals/scheduler/scheduler.proto:77]:
```suggestion
    string subnet = 3;
}
```

File: docs/proposals/scheduler/scheduler.proto [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+message Instance {
+    string id = 1;
+    string name = 2;
+    Type type = 3;
+    State state = 4;
+    string uri = 5;
+    [] string environnement = 6;
+    Resource resource = 7;
+    [] string ports = 8;
+}
```
Comment (Alexis-Bernard) [inline docs/proposals/scheduler/scheduler.proto:38]: You need to add ip address for an instance
```suggestion
    string ip = 9;
}
```

**Review by Alexis-Bernard — CHANGES_REQUESTED** [review]
You need to add ip adress for an new node and a instance

**Review by Alexis-Bernard — APPROVED** [review]
Good job !

**Pull request merged**